recipes/soci/all/conanfile.py

```diff
@@ -0,0 +1,187 @@
+            self.requires("oracle/x.y.z")
+
+        if self.options.with_boost:
+            self.requires("boost/1.73.0")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("soci-{}".format(self.version), self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+
+        self._cmake.definitions["BINDIR"] = os.path.join(self.package_folder, "bin")
+        self._cmake.definitions["LIBDIR"] = os.path.join(self.package_folder, "lib")
+        self._cmake.definitions["LIB_SUFFIX"] = ""
+
+        self._cmake.definitions["SOCI_SHARED"] = self.options.shared
+        self._cmake.definitions["SOCI_STATIC"] = not self.options.shared
+        self._cmake.definitions["SOCI_TESTS"] = False
+
+        self._cmake.definitions["WITH_BOOST"] = self.options.with_boost
+        self._cmake.definitions["SOCI_EMPTY"] = self.options.with_backend_empty
+        self._cmake.definitions["WITH_SQLITE3"] = self.options.with_backend_sqlite3
+        self._cmake.definitions["WITH_MYSQL"] = self.options.with_backend_mysql
+        self._cmake.definitions["WITH_ODBC"] = self.options.with_backend_odbc
+        self._cmake.definitions["WITH_POSTGRESQL"] = self.options.with_backend_postgresql
+
+        self._cmake.definitions["WITH_DB2"] = self.options.with_backend_db2
+        self._cmake.definitions["WITH_FIREBIRD"] = self.options.with_backend_firebird
+        self._cmake.definitions["WITH_ORACLE"] = self.options.with_backend_oracle
+
+        self._cmake.configure()
+        return self._cmake
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+        if self.options.with_backend_sqlite3:
+            tools.replace_in_file("FindSQLite3.cmake", "SQLite3_FOUND", "SQLITE3_FOUND")
+            tools.replace_in_file("FindSQLite3.cmake", "SQLite3_INCLUDE_DIRS", "SQLITE3_INCLUDE_DIRS")
+            tools.replace_in_file("FindSQLite3.cmake", "SQLite3_LIBRARIES", "SQLITE3_LIBRARIES")
+        if self.options.with_backend_mysql:
+            os.rename("Findlibmysqlclient.cmake", "FindMySQL.cmake")
+            tools.replace_in_file("FindMySQL.cmake", "libmysqlclient_FOUND", "MYSQL_FOUND")
+            tools.replace_in_file("FindMySQL.cmake", "libmysqlclient_INCLUDE_DIRS", "MYSQL_INCLUDE_DIRS")
+            tools.replace_in_file("FindMySQL.cmake", "libmysqlclient_LIBRARIES", "MYSQL_LIBRARIES")
+        if self.options.with_backend_postgresql:
+            tools.replace_in_file("FindPostgreSQL.cmake", "PostgreSQL_FOUND", "POSTGRESQL_FOUND")
+            tools.replace_in_file("FindPostgreSQL.cmake", "PostgreSQL_INCLUDE_DIRS", "POSTGRESQL_INCLUDE_DIRS")
+            tools.replace_in_file("FindPostgreSQL.cmake", "PostgreSQL_LIBRARIES", "POSTGRESQL_LIBRARIES")
```

Comment:
@danimtb @jgsogo @uilianries 
This feels like a hack.
Do there exist hooks to change the behaviour of generators?

Reply:
Indeed it's a hack, but we don't a hook for it.

Reply:
Yes, we should apply a patch to `soci` to take into account these casing issues... probably we should have a look to existing CMake modules for these libraries and send the patch upstream too.

@madebr there are no hooks to change the behavior of existing generators, you can write your own, but not change existing ones.

Reply:
Done